src/Text/Pandoc/Class/PandocMonad.hs

```diff
@@ -586,6 +587,23 @@ readDataFile fname = do
             then readFileStrict (userDir </> fname)
             else readDefaultDataFile fname
 
+-- | Read metadata file from the working directory or, if not found there, from
+-- the metadata subdirectory of the user data directory.
+readMetadataFile :: PandocMonad m => FilePath -> m B.ByteString
+readMetadataFile fname = do
+  existsInWorkingDir <- fileExists fname
+  if existsInWorkingDir then readFileStrict fname
+  else do
+    dataDir <- getUserDataDir
+    case dataDir of
+      Nothing ->
+        throwError $ PandocCouldNotFindDataFileError $ T.pack fname
+      Just userDir -> do
+        let path = userDir </> "metadata" </> fname
+        existsInUserDir <- fileExists path
+        if existsInUserDir then readFileStrict path
+        else throwError $ PandocCouldNotFindDataFileError $ T.pack fname
```

Comment:
This isn't really the appropriate error since it's not a data file.

Reply:
I could add `PandocCouldNotFindMetadataFileError`? That would be consistent with `readDataFile` and `readMetadataFile`.

Reply:
Added `PandocCouldNotFindMetadataFileError`.

Reply:
Note that a missing metadata file specified with `--metadata-dir` results in the more general `PandocIOError` (via `readFileStrict`) in `master`. This PR changes that.